targets/TARGET_Samsung/TARGET_SIDK_S5JS100/serial_api.c

```diff
@@ -0,0 +1,570 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#if DEVICE_SERIAL
+
+// math.h required for floating point operations for baud rate calculation
+#include <math.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "serial_api.h"
+#include "cmsis.h"
+#include "pinmap.h"
+#include "PinNames.h"
+#include "mbed_error.h"
+#include "gpio_api.h"
+
+#define USI_PTR(ptr) ((S5JS100_USI_UART_TypeDef *)(ptr))
+#define UART_PTR(ptr)	((S5JS100_UART_TypeDef *)(ptr))
+
+/******************************************************************************
+ * INITIALIZATION
+ ******************************************************************************/
+
+static const PinMap PinMap_UART_TX[] = {
+	{UART_TX0, UART_0, 0},
+	{UART_TX1, UART_1, 0},
+	{UART_TX2, UART_2, 0},
+	{UART_TX3, UART_3, 0},
+	{NC, NC, 0}
+};
+static const PinMap PinMap_UART_RX[] = {
+	{UART_RX0, UART_0, 0},
+	{UART_RX1, UART_1, 0},
+	{UART_RX2, UART_2, 0},
+	{UART_RX3, UART_3, 0},
+	{NC, NC, 0}
+};
+#define UART_NUM 4
+
+static uart_irq_handler irq_handler[UART_NUM];
+
+int stdio_uart_inited = 0;
+serial_t stdio_uart;
+
+struct serial_global_data_s {
+	uint32_t serial_irq_id;
+	gpio_t sw_rts, sw_cts;
+	uint8_t count, rx_irq_set_flow, rx_irq_set_api;
+};
+
+static struct serial_global_data_s uart_data[UART_NUM];
+
+void serial_init(serial_t *obj, PinName tx, PinName rx)
+{
+	int is_stdio_uart = 0;
+
+	// determine the UART to use
+	UARTName uart_tx = (UARTName)pinmap_peripheral(tx, PinMap_UART_TX);
+	UARTName uart_rx = (UARTName)pinmap_peripheral(rx, PinMap_UART_RX);
+	UARTName uart = (UARTName)pinmap_merge(uart_tx, uart_rx);
+	if ((int)uart == NC) {
```

Comment:
I've noticed coding style differences in HAL files like spi_api.c or serial_api.c - using 8 spaces and similar. Can you run astyle on these files please? it would be good to have HAL implementation following the coding style if we are adding new files